File: Tokens/bikemarket/contracts/src/test/java/net/corda/samples/bikemarket/contracts/StateTests.java
package net.corda.samples.bikemarket.contracts;

import net.corda.samples.bikemarket.states.WheelsTokenState;
import net.corda.testing.node.MockServices;
import org.junit.Test;
public class StateTests {
    private final MockServices ledgerServices = new MockServices();

    //sample State tests
    @Test
    public void hasSerialNumFieldOfCorrectType() throws NoSuchFieldException {
        // Does the message field exist?
        WheelsTokenState.class.getDeclaredField("serialNum");
        // Is the message field of the correct type?
        assert(WheelsTokenState.class.getDeclaredField("serialNum").getType().equals(String.class));
    }
}

File: kotlin-samples/Tokens/bikemarket/contracts/src/test/kotlin/net/corda/samples/bikemarket/contracts/StateTests.kt
package net.corda.samples.bikemarket.contracts

import net.corda.samples.bikemarket.states.WheelsTokenState
import net.corda.testing.node.MockServices
import org.junit.Test

class StateTests {
    private val ledgerServices = MockServices()

    //sample State tests
    @Test
    fun hasSerialNumFieldOfCorrectType() {
        // Does the message field exist?
        WheelsTokenState::class.java.getDeclaredField("serialNum")
        // Is the message field of the correct type?
        assert(WheelsTokenState::class.java.getDeclaredField("serialNum").type == String::class.java)
    }
}